fabric-common/lib/Utils.js

```diff
@@ -622,3 +622,43 @@ module.exports.randomize = (items) => {
 module.exports.checkParameter = (name) => {
 	throw Error(`Missing ${name} parameter`);
 };
+
+/**
+ * Map CSRUtil.newCSRPEM style extensions:
+ * ```
+ * {
+ *     subjectAltName: {
+ *         array: [...],
+ *     },
+ * }
+ * ```
+ *
+ * to CertificationRequest style extensions:
+ * ```
+ * {
+ *     extname: 'subjectAltName',
+ *     array: [...],
+ * }
+ * ```
+ * @private
+ */
+module.exports.mapCSRExtensions = (extensions) => {
+	if (!Array.isArray(extensions)) {
+		return extensions;
+	}
+
+	const results = [];
+	extensions.forEach(extension => {
+		const isCertificationRequestExtension = typeof extension.extname === 'string';
+		if (isCertificationRequestExtension) {
+			results.push(extension);
+		} else {
+			Object.entries(extension).forEach(([extname, props]) => {
+				const extensionRequest = Object.assign({}, props, {extname});
+				results.push(extensionRequest);
+			});
+		}
+	});
+
+	return results;
+};
```

Comment:
I understand this method is helpful to keep backwards compatibility. However, I imagine that eventually, in a major release it could or should disappear because IMO it is important to maintain same `jsrsasign` data structure. Unfortunately, the project is not using TS which makes it difficult to know what's the extension's data structure.

Reply:
jsrsasign is a classic project for a while, and almost there is hardly a chance to migrate to TS or ECMA6 module.
@inglkruiz Or do you mean you suggest we can consider migrating fabric-common to a TS based project?

Reply:
@inglkruiz this should allow both the backwards compatible and new syntax to be used but you're right, we could remove the backwards compatible support in a future release.

I agree that TypeScript would be much simpler if the implementation had been done in TypeScript rather than having type definitions hand-crafted after the fact. You'll see that I eventually managed to get the whole implementation of the **fabric-network** package over to TypeScript, and the new [fabric-gateway](https://github.com/hyperledger/fabric-gateway) client API for Fabric v2.4 started with TypeScript from day one for exactly this reason.

Reply:
@davidkhala To be precise, yes, I'd suggest migrating to a TS based project. IMO using Nx Monorepo tool could help you to avoid the hassle of configuring the project (It's what I've been using for this past 3-4 years) and the SDK folders could be just publishable libraries (I have not work with that feature yet) https://nx.dev/l/r/structure/buildable-and-publishable-libraries